Library/Homebrew/cmd/upgrade.rb

```diff
@@ -5,10 +5,17 @@ module Homebrew
   def upgrade
     Homebrew.perform_preinstall_checks
 
-    if ARGV.named.empty?
+    if ARGV.include?("--all") #|| ARGV.named.empty?
```

Comment:
I think it should be `if ARGV.include?("--all") || ARGV.named.empty?` for the time being.